demo/midgard-node/src/fibers/fetch-and-insert-deposit-utxos.ts

```diff
@@ -0,0 +1,92 @@
+import * as SDK from "@al-ft/midgard-sdk";
+import { Console, Effect, Ref } from "effect";
+import {
+  AlwaysSucceedsContract,
+  Globals,
+  Lucid,
+  NodeConfig,
+} from "@/services/index.js";
+import { LucidEvolution, CML, Data, fromHex } from "@lucid-evolution/lucid";
+import { DepositsDB } from "@/database/index.js";
+import { SqlClient } from "@effect/sql";
+import { DatabaseError } from "@/database/utils/common.js";
+import { Schedule } from "effect";
+
+const fetchDepositUTxOs = (
+  lucid: LucidEvolution,
+  inclusionStartTime: number,
+  inclusionEndTime: number,
+): Effect.Effect<
+  SDK.TxBuilder.Deposit.DepositUTxO[],
+  | SDK.Utils.LucidError
+  | SDK.Utils.DataCoercionError
+  | SDK.Utils.AssetError
+  | SDK.Utils.UnauthenticUtxoError,
+  AlwaysSucceedsContract | NodeConfig
+> =>
+  Effect.gen(function* () {
+    const { depositAuthValidator } = yield* AlwaysSucceedsContract;
+    const fetchConfig: SDK.TxBuilder.Deposit.FetchConfig = {
+      depositAddress: depositAuthValidator.spendScriptAddress,
+      depositPolicyId: depositAuthValidator.policyId,
+      inclusionStartTime: BigInt(inclusionStartTime),
+      inclusionEndTime: BigInt(inclusionEndTime),
+    };
+    return yield* SDK.Endpoints.fetchDepositUTxOsProgram(lucid, fetchConfig);
+  });
+
+export const fetchAndInsertDepositUTxOs = (): Effect.Effect<
+  void,
+  | SDK.Utils.LucidError
+  | SDK.Utils.DataCoercionError
+  | SDK.Utils.AssetError
+  | SDK.Utils.UnauthenticUtxoError
+  | DatabaseError,
+  AlwaysSucceedsContract | NodeConfig | Lucid | SqlClient.SqlClient | Globals
```

Comment:
Let's be consistent in use of our `Database` alias.
```suggestion
  AlwaysSucceedsContract | NodeConfig | Lucid | Database | Globals
```